### PR TITLE
remove dump_test.go

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve? 

`dump_test.go` is too hard to maintain because 
- we need to match every query and keep the order
- we need to assign a expected result(which is not easy to write correctly) for every query

So even a tiny change may break this test and it takes a lot of time to figure it out.

The work done by `dump_test.go` is much like "integration test", we should put them to a real integration test.

### What is changed and how it works?

Remove `dump_test.go` and add an integration test.

### Check List 

Tests 

 - Unit test
 - Integration test

Side effects

N/A

Related changes

 - Need to cherry-pick to the release branch
 
### Release note

No release note

### CODE REVIEW

File: tests/basic/run.sh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# Test for simple case.
+run_sql "drop database if exists \`$DB_NAME\`;"
+run_sql "create database \`$DB_NAME\`;"
+run_sql "create table \`$DB_NAME\`.\`$TABLE_NAME\` (a int);"
+run_sql "insert into \`$DB_NAME\`.\`$TABLE_NAME\` values (1), (2);"
+
+run_dumpling
+
+cnt=`grep -w "(.*)" ${DUMPLING_OUTPUT_DIR}/${DB_NAME}.${TABLE_NAME}.000000000.sql|wc -l`
+echo "records count is ${cnt}"
```
Comment (lichunzhu) [inline tests/basic/run.sh:18]: Should we check the content here?

Comment (tangenta) [reply tests/basic/run.sh:18]: Updated.

**Comment by lichunzhu** [comment]
Should we delete this unit test? @kennytm

**Review by kennytm — COMMENTED** [review]
lgtm

**Review by kennytm — APPROVED** [review]

**Pull request merged**